### ISSUE TITLE
Optimize Salmon index length calculation

### ISSUE THREAD

**Issue by kurtwheeler** [AlexsLemonade/refinebio#157]
### New Issue Checklist
- [x] The title is short and descriptive
- [x] The issue contains an:
  - [ ] Idea (new feature, user story, etc)
  - [X] Problem
- [x] You have explained the:
  - [x] Context
  - [x] Problem or idea
  - [x] Solution or next step
  
### Context

Depending on the average transcript read length for FASTA files, we should use either a salmon index that is optimized for long or short reads. Therefore when running salmon we first calculate the average read length and then run salmon. However we are doing this with Python and it is single-core gunziping the FASTA files line by line. 

### Problem or idea

@Miserlou  thinks there is a faster way of doing this, maybe by shelling and using UNIX! It 'd be faster to gunzip with multicores and then count lines that way, ex:
```bash
$ pigz file.fasq.gz;
$ cat file.fastq | echo $((`wc -l`/4))
```

### Solution or next step

Try shelling out and using multiple cores to speed up the average read length calculation.

**Comment by jaclyn-taroni** [comment]
Salmon assumes that the reads you give it are in a random order -- see the first **Note** [here](https://salmon.readthedocs.io/en/latest/salmon.html#using-salmon) in the Salmon docs. Do you think it would behoove us to shuffle the `fastq.gz` file around the same time we calculate the read lengths just in case they are ordered in some way? (This is just something that has been in the back of my mind that might be worth discussing -- can file another ticket, let me know.)

**Comment by Miserlou** [comment]
This ticket is related to a comment around a single line of code that has to do with how unzipping in Python is slow. If your idea is something we should investigate @jaclyn-taroni , I think it'd be best to open a new ticket.

**Comment by jaclyn-taroni** [comment]
I believe this was closed via #339
